### PR TITLE
issue #2621 - add overloaded builder setters

### PR DESCRIPTION
**Pull request by lmsurpre**
Draft pull request to see how we like this...
See https://github.com/IBM/FHIR/blob/a7e6ad81c1f723a5541a83088be4993e1ff76508/fhir-model/src/test/java/com/ibm/fhir/model/test/ModelBuilderTest.java#L32-L50 for a quick overview of what this allows.

Note:  We probably need to update fhir-example-generator to have it handle all these new setter methods correctly.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRPatchTest.java [hunk 1]
```diff
@@ -112,7 +113,7 @@ public void testJSONPatchRemoveOperation() throws Exception {
 
         // create a copy of the patient and update it using the model API
         Patient.Builder patientBuilder = patient.toBuilder();
-        patientBuilder.active(null);
```
Comment (lmsurpre) [inline fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRPatchTest.java:115]: an unfortunate consequence of adding the overloads

File: fhir-model/src/test/java/com/ibm/fhir/model/test/ModelBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Test(expectedExceptions = IllegalStateException.class)
+    public void testVarArgWithNull() {
+        Basic.builder()
+            .code(CodeableConcept.builder()
+                .coding((Coding)null)
+                .build())
+            .build();
+    }
+
+    // This one throws a NullPointerException rather than the expected IllegalStateException...is that an issue?
```
Comment (lmsurpre) [inline fhir-model/src/test/java/com/ibm/fhir/model/test/ModelBuilderTest.java:82]: is this an issue?

Comment (lmsurpre) [reply fhir-model/src/test/java/com/ibm/fhir/model/test/ModelBuilderTest.java:82]: personally I think its fine to throw the NPE in this case, but maybe we could add it to the javadoc

**Review by JohnTimm — APPROVED** [review]
LGTM

**Pull request merged**